In [ ]:
import pandas as pd

In [ ]:
import numpy as np

# importation des fichiers CSV

In [ ]:
population = pd.read_csv ('population_original.csv')
aide_alimentaire = pd.read_csv('aide_alimentaire.csv')
sous_nutrition = pd.read_csv ('sous_nutrition.csv')
dispo_alimentaire = pd.read_csv('dispo_alimentaire.csv')
population2017= pd.read_csv('population_2017.csv')

In [ ]:
population.head()

Zone  Année     Valeur
0  Afghanistan   2013  32269.589
1  Afghanistan   2014  33370.794
2  Afghanistan   2015  34413.603
3  Afghanistan   2016  35383.032
4  Afghanistan   2017  36296.113

In [ ]:
population.describe(include='all')

Zone        Année        Valeur
count          1416  1416.000000  1.416000e+03
unique          236          NaN           NaN
top     Afghanistan          NaN           NaN
freq              6          NaN           NaN
mean            NaN  2015.500000  3.144793e+04
std             NaN     1.708428  1.300812e+05
min             NaN  2013.000000  7.930000e-01
25%             NaN  2014.000000  3.783410e+02
50%             NaN  2015.500000  5.126480e+03
75%             NaN  2017.000000  1.930666e+04
max             NaN  2018.000000  1.427648e+06

In [ ]:
population_annee = population.pivot_table(index = ['Zone'],
                columns = 'Année', values = 'Valeur', aggfunc = sum).reset_index()
population_annee.head()

Année            Zone       2013       2014       2015       2016       2017  \
0         Afghanistan  32269.589  33370.794  34413.603  35383.032  36296.113   
1      Afrique du Sud  53687.121  54544.186  55386.367  56207.646  57009.756   
2             Albanie   2903.790   2896.305   2890.513   2886.438   2884.169   
3             Algérie  38140.133  38923.692  39728.025  40551.392  41389.189   
4           Allemagne  81174.367  81450.378  81787.411  82193.768  82658.409   

Année       2018  
0      37171.921  
1      57792.518  
2       2882.740  
3      42228.408  
4      83124.418

In [ ]:
population2017.drop(population2017[population2017.Zone.str.startswith("Chine,")==True].index, inplace=True)
population2017.drop(population2017[population2017.Zone.str.startswith("Chine -")==True].index, inplace=True)

**La population mondiale est de 7 490 611 220 habitants**

In [ ]:
population_mondiale = (population2017['Valeur']*1000).sum()
population_mondiale

7490611220.0

# **Quelle est la proportion de personnes en état de sous-nutrition ?**

In [ ]:
sous_nutrition.drop(sous_nutrition[sous_nutrition.Zone.str.startswith("Chine,")==True].index, inplace=True)
sous_nutrition.drop(sous_nutrition[sous_nutrition.Zone.str.startswith("Chine -")==True].index, inplace=True)
sous_nutrition.head()

Zone      Année Valeur
0  Afghanistan  2012-2014    8.6
1  Afghanistan  2013-2015    8.8
2  Afghanistan  2014-2016    8.9
3  Afghanistan  2015-2017    9.7
4  Afghanistan  2016-2018   10.5

In [ ]:
sousalimentation_annee = sous_nutrition.pivot_table(index = ['Zone'],
                        columns = 'Année', values = 'Valeur', aggfunc = sum).reset_index()
sousalimentation_annee.head(10)

Année                Zone 2012-2014 2013-2015 2014-2016 2015-2017 2016-2018  \
0             Afghanistan       8.6       8.8       8.9       9.7      10.5   
1          Afrique du Sud       2.2       2.5       2.8         3       3.1   
2                 Albanie       0.1       0.1       0.1       0.1       0.1   
3                 Algérie       1.3       1.3       1.3       1.3       1.3   
4               Allemagne         0         0         0         0         0   
5                 Andorre         0         0         0         0         0   
6                  Angola       7.6       6.2       5.3       5.6       5.8   
7      Antigua-et-Barbuda         0         0         0         0         0   
8         Arabie saoudite       1.5       1.5       1.5       1.5       1.6   
9               Argentine       1.2         0       1.1       1.3       1.5   

Année 2017-2019  
0          11.1  
1           3.3  
2           0.1  
3           1.2  
4             0  
5             0  
6           5.7  
7             0  
8           1.6  
9           1.7

In [ ]:
sousalimentation_annee['2016-2018'].unique()

array(['10.5', '3.1', '0.1', '1.3', 0, '5.8', '1.6', '1.5', '<0.1',
       '21.5', '1.7', '0.5', '0.2', '3.6', '0.8', '2.4', '0.6', '2.7',
       '1.4', '4.9', '0.3', '2', '5.3', '190.1', '23.6', '3.9', '9',
       '11.9', '0.4', '1.8', '1', '3.3', '8.4', '9.4', '7.3', '1.1',
       '22.8', '0.9', '24.8', '15.7', '2.2', '4.2', '12', '13.4', '5',
       '5.7', '6.2', '8', '6.5', '4.6', '21.1'], dtype=object)

In [ ]:
sousalimentation_annee.loc[sousalimentation_annee['2016-2018'] == '<0.1'] = 0

In [ ]:
sousalimentation_annee['2016-2018'] = sousalimentation_annee['2016-2018'].apply(pd.to_numeric)

In [ ]:
population_sousnutrie = sousalimentation_annee['2016-2018'].sum()
print(int(population_sousnutrie*1000000),'de personnes sous-nutries pour 2017')

534799999 de personnes sous-nutries pour 2017


In [ ]:
ratio_sousnutries = ((population_sousnutrie * 1000000) / population_mondiale)*100
ratio_sousnutries
print('La proportion de la population mondiale considérée comme étant en sous-nutrition pour l\'année 2017 est de',round(ratio_sousnutries,3),'%.')

La proportion de la population mondiale considérée comme étant en sous-nutrition pour l'année 2017 est de 7.14 %.


In [ ]:
dispo_alimentaire.head()

Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    NaN   
1  Afghanistan        Agrumes, Autres  vegetale                    NaN   
2  Afghanistan  Aliments pour enfants  vegetale                    NaN   
3  Afghanistan                 Ananas  vegetale                    NaN   
4  Afghanistan                Bananes  vegetale                    NaN   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  NaN                                             5.0   
1                  NaN                                             1.0   
2                  NaN                                             1.0   
3                  NaN                                             0.0   
4                  NaN                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.72        
1                                               1.29        
2                                               0.06        
3                                               0.00        
4                                               2.70        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.20               
1                                               0.01               
2                                               0.01               
3                                                NaN               
4                                               0.02               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.77          
1                                               0.02          
2                                               0.03          
3                                                NaN          
4                                               0.05          

   Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                      53.0                      NaN                      NaN   
1                      41.0                      2.0                     40.0   
2                       2.0                      NaN                      2.0   
3                       0.0                      NaN                      0.0   
4                      82.0                      NaN                     82.0   

   Nourriture  Pertes  Production  Semences  Traitement  Variation de stock  
0        53.0     NaN        53.0       NaN         NaN                 NaN  
1        39.0     2.0         3.0       NaN         NaN                 NaN  
2         2.0     NaN         NaN       NaN         NaN                 NaN  
3         0.0     NaN         NaN       NaN         NaN                 NaN  
4        82.0     NaN         NaN       NaN         NaN                 NaN

In [ ]:
bilan_alimentaire = pd.merge(dispo_alimentaire, population, on = 'Zone')
bilan_alimentaire.columns

Index(['Zone', 'Produit', 'Origine', 'Aliments pour animaux',
       'Autres Utilisations', 'Disponibilité alimentaire (Kcal/personne/jour)',
       'Disponibilité alimentaire en quantité (kg/personne/an)',
       'Disponibilité de matière grasse en quantité (g/personne/jour)',
       'Disponibilité de protéines en quantité (g/personne/jour)',
       'Disponibilité intérieure', 'Exportations - Quantité',
       'Importations - Quantité', 'Nourriture', 'Pertes', 'Production',
       'Semences', 'Traitement', 'Variation de stock', 'Année', 'Valeur'],
      dtype='object')

In [ ]:
bilan_alimentaire 

Zone           Produit   Origine  Aliments pour animaux  \
0       Afghanistan  Abats Comestible   animale                    NaN   
1       Afghanistan  Abats Comestible   animale                    NaN   
2       Afghanistan  Abats Comestible   animale                    NaN   
3       Afghanistan  Abats Comestible   animale                    NaN   
4       Afghanistan  Abats Comestible   animale                    NaN   
...             ...               ...       ...                    ...   
92491  Îles Salomon    Épices, Autres  vegetale                    NaN   
92492  Îles Salomon    Épices, Autres  vegetale                    NaN   
92493  Îles Salomon    Épices, Autres  vegetale                    NaN   
92494  Îles Salomon    Épices, Autres  vegetale                    NaN   
92495  Îles Salomon    Épices, Autres  vegetale                    NaN   

       Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                      NaN                                             5.0   
1                      NaN                                             5.0   
2                      NaN                                             5.0   
3                      NaN                                             5.0   
4                      NaN                                             5.0   
...                    ...                                             ...   
92491                  NaN                                             4.0   
92492                  NaN                                             4.0   
92493                  NaN                                             4.0   
92494                  NaN                                             4.0   
92495                  NaN                                             4.0   

       Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                                   1.72        
1                                                   1.72        
2                                                   1.72        
3                                                   1.72        
4                                                   1.72        
...                                                  ...        
92491                                               0.48        
92492                                               0.48        
92493                                               0.48        
92494                                               0.48        
92495                                               0.48        

       Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                                   0.20               
1                                                   0.20               
2                                                   0.20               
3                                                   0.20               
4                                                   0.20               
...                                                  ...               
92491                                               0.21               
92492                                               0.21               
92493                                               0.21               
92494                                               0.21               
92495                                               0.21               

       Disponibilité de protéines en quantité (g/personne/jour)  \
0                                                   0.77          
1                                                   0.77          
2                                                   0.77          
3                                                   0.77          
4                                                   0.77          
...                                                  ...          
92491                                               0.15          
92492  

In [ ]:
bilan_alimentaire['Disponibilité alimentaire (Kcal/an)'] = bilan_alimentaire['Disponibilité alimentaire (Kcal/personne/jour)'] \
* 365 * bilan_alimentaire['Valeur'] 
bilan_alimentaire['Disponibilité intérieur (KG)'] = bilan_alimentaire['Disponibilité intérieure']*1000000


In [ ]:
bilan_alimentaire['Disponibilité de protéines en quantité (Kg/an)']= bilan_alimentaire['Disponibilité de protéines en quantité (g/personne/jour)']\
* 365 * .001 * bilan_alimentaire['Valeur']

bilan_alimentaire['Disponibilité alimentaire en quantité (Kg/an)'] = bilan_alimentaire['Disponibilité alimentaire en quantité (kg/personne/an)'] \
* bilan_alimentaire['Valeur']

In [ ]:
bilan_alimentaire['Ratio énergie/poids (Kcal/Kg)'] = bilan_alimentaire['Disponibilité alimentaire (Kcal/an)'] \
/ bilan_alimentaire['Disponibilité alimentaire en quantité (Kg/an)']
bilan_alimentaire = bilan_alimentaire.replace([np.inf, -np.inf], np.nan).dropna(subset=["Ratio énergie/poids (Kcal/Kg)"], how="all")


#**Question : Combien d'humains pourraient être nourris avec la disponibilité alimentaire mondiale ?**

In [ ]:
filt_bilan_alimentaire =bilan_alimentaire.query ('Année ==2017')
filt_bilan_alimentaire.head(20)

Zone                  Produit   Origine  Aliments pour animaux  \
4    Afghanistan         Abats Comestible   animale                    NaN   
10   Afghanistan          Agrumes, Autres  vegetale                    NaN   
16   Afghanistan    Aliments pour enfants  vegetale                    NaN   
28   Afghanistan                  Bananes  vegetale                    NaN   
34   Afghanistan             Beurre, Ghee   animale                    NaN   
40   Afghanistan                    Bière  vegetale                    NaN   
46   Afghanistan                      Blé  vegetale                    NaN   
70   Afghanistan                    Crème   animale                    NaN   
82   Afghanistan                   Dattes  vegetale                    NaN   
88   Afghanistan       Edulcorants Autres  vegetale                    NaN   
94   Afghanistan            Feve de Cacao  vegetale                    NaN   
100  Afghanistan           Fruits, Autres  vegetale                    NaN   
118  Afghanistan   Graisses Animales Crue   animale                    NaN   
124  Afghanistan  Huil Plantes Oleif Autr  vegetale                    NaN   
130  Afghanistan   Huile Graines de Coton  vegetale                    NaN   
142  Afghanistan            Huile d'Olive  vegetale                    NaN   
148  Afghanistan  Huile de Colza&Moutarde  vegetale                    NaN   
154  Afghanistan           Huile de Palme  vegetale                    NaN   
160  Afghanistan            Huile de Soja  vegetale                    NaN   
166  Afghanistan          Huile de Sésame  vegetale                    NaN   

     Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
4                    NaN                                             5.0   
10                   NaN                                             1.0   
16                   NaN                                             1.0   
28                   NaN                                             4.0   
34                   NaN                                            23.0   
40                   NaN                                             0.0   
46                   NaN                                          1369.0   
70                   NaN                                             3.0   
82                   NaN                                             0.0   
88                   NaN                                             2.0   
94                   NaN                                             0.0   
100                  NaN                                             9.0   
118                  NaN                                            12.0   
124                359.0                                             2.0   
130                  1.0                                             3.0   
142                  NaN                                             0.0   
148                  NaN                                             2.0   
154                 55.0                                            51.0   
160                  NaN                                            13.0   
166                  NaN                                             1.0   

     Disponibilité alimentaire en quantité (kg/personne/an)  \
4                                                 1.72        
10                                                1.29        
16                                                0.06        
28                                                2.70        
34                                                1.17        
40                                                0.09        
46                                              160.23        
70                                                0.60        
82                                                0.02        
88                                                0.53        
94                                                0.02        
100                 

In [ ]:
filt_bilan_alimentaire1 =filt_bilan_alimentaire['Disponibilité alimentaire (Kcal/an)'].sum()
filt_bilan_alimentaire1

7631922367877.385

#la disponibilité alimentaire (Kcla/an) mondiale est de 7631922367877.385


In [ ]:
dispo_alimentaire=pd.read_csv('dispo_alimentaire.csv')

In [ ]:
dispo_alimentaire.head()

Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    NaN   
1  Afghanistan        Agrumes, Autres  vegetale                    NaN   
2  Afghanistan  Aliments pour enfants  vegetale                    NaN   
3  Afghanistan                 Ananas  vegetale                    NaN   
4  Afghanistan                Bananes  vegetale                    NaN   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  NaN                                             5.0   
1                  NaN                                             1.0   
2                  NaN                                             1.0   
3                  NaN                                             0.0   
4                  NaN                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.72        
1                                               1.29        
2                                               0.06        
3                                               0.00        
4                                               2.70        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.20               
1                                               0.01               
2                                               0.01               
3                                                NaN               
4                                               0.02               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.77          
1                                               0.02          
2                                               0.03          
3                                                NaN          
4                                               0.05          

   Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                      53.0                      NaN                      NaN   
1                      41.0                      2.0                     40.0   
2                       2.0                      NaN                      2.0   
3                       0.0                      NaN                      0.0   
4                      82.0                      NaN                     82.0   

   Nourriture  Pertes  Production  Semences  Traitement  Variation de stock  
0        53.0     NaN        53.0       NaN         NaN                 NaN  
1        39.0     2.0         3.0       NaN         NaN                 NaN  
2         2.0     NaN         NaN       NaN         NaN                 NaN  
3         0.0     NaN         NaN       NaN         NaN                 NaN  
4        82.0     NaN         NaN       NaN         NaN                 NaN

In [ ]:
total=dispo_alimentaire['Disponibilité intérieure'].sum()
total

9848994.0

In [ ]:
perte= (dispo_alimentaire['Pertes'].sum()/total)*100
perte

4.606541541196999

In [ ]:
aliment= (dispo_alimentaire['Aliments pour animaux'].sum()/total)*100
aliment

13.242418464261426

In [ ]:
nourriture=(dispo_alimentaire['Nourriture'].sum()/total)*100
nourriture

49.51021393657058

In [ ]:
dispo_humain = round ((filt_bilan_alimentaire1 / 912500),2)
dispo_humain

8363750.54

**La disponibilité alimentaire pourrait nourrir 8 milliards d'habitants**

In [ ]:
pourcentage_humain_nourris_dispoalim_cal = round((dispo_humain*1000 / population_mondiale * 100),2)
pourcentage_humain_nourris_dispoalim_cal

111.66

**La part d'humain que l'on peut nourrir avec la disponibilité alimentaire est de 111% de la population**


#**Question : Combien d'humains pourraient être nourris si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture ?**



In [ ]:
tab_vegetal=filt_bilan_alimentaire
tab_vegetal = tab_vegetal[tab_vegetal['Origine'] == 'vegetale']
tab_vegetal.head()

Zone                Produit   Origine  Aliments pour animaux  \
10  Afghanistan        Agrumes, Autres  vegetale                    NaN   
16  Afghanistan  Aliments pour enfants  vegetale                    NaN   
28  Afghanistan                Bananes  vegetale                    NaN   
40  Afghanistan                  Bière  vegetale                    NaN   
46  Afghanistan                    Blé  vegetale                    NaN   

    Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
10                  NaN                                             1.0   
16                  NaN                                             1.0   
28                  NaN                                             4.0   
40                  NaN                                             0.0   
46                  NaN                                          1369.0   

    Disponibilité alimentaire en quantité (kg/personne/an)  \
10                                               1.29        
16                                               0.06        
28                                               2.70        
40                                               0.09        
46                                             160.23        

    Disponibilité de matière grasse en quantité (g/personne/jour)  \
10                                               0.01               
16                                               0.01               
28                                               0.02               
40                                                NaN               
46                                               4.69               

    Disponibilité de protéines en quantité (g/personne/jour)  \
10                                               0.02          
16                                               0.03          
28                                               0.05          
40                                               0.00          
46                                              36.91          

    Disponibilité intérieure  ...  Semences  Traitement  Variation de stock  \
10                      41.0  ...       NaN         NaN                 NaN   
16                       2.0  ...       NaN         NaN                 NaN   
28                      82.0  ...       NaN         NaN                 NaN   
40                       3.0  ...       NaN         NaN                 NaN   
46                    5992.0  ...     322.0         NaN              -350.0   

    Année     Valeur  Disponibilité alimentaire (Kcal/an)  \
10   2017  36296.113                         1.324808e+07   
16   2017  36296.113                         1.324808e+07   
28   2017  36296.113                         5.299232e+07   
40   2017  36296.113                         0.000000e+00   
46   2017  36296.113                         1.813662e+10   

    Disponibilité intérieur (KG)  \
10                  4.100000e+07   
16                  2.000000e+06   
28                  8.200000e+07   
40                  3.000000e+06   
46                  5.992000e+09   

    Disponibilité de protéines en quantité (Kg/an)  \
10                                      264.961625   
16                                      397.442437   
28                                      662.404062   
40                                        0.000000   
46                                   488986.678753   

    Disponibilité alimentaire en quantité (Kg/an)  \
10                                   4.682199e+04   
16                                   2.177767e+03   
28                                   9.799951e+04   
40                                   3.266650e+03   
46                                   5.815726e+06   

    Ratio énergie/poids (Kcal/Kg)  
10                     282.945736  
16                    6083.333333  
28                     540.740741  
40                       0.000000  
46                    3118.548337  

[5 rows x 25 col

In [ ]:
tab_vegetal.groupby('Produit').sum().reset_index()[['Produit','Disponibilité alimentaire (Kcal/an)']]

Produit  Disponibilité alimentaire (Kcal/an)
0          Agrumes, Autres                         2.639647e+09
1    Aliments pour enfants                         2.542592e+09
2                   Ananas                         7.331545e+09
3   Arachides Decortiquees                         6.751420e+10
4                   Avoine                         6.622216e+09
..                     ...                                  ...
67                  Sésame                         8.981478e+09
68                     Thé                         2.211801e+09
69                 Tomates                         2.792788e+10
70                     Vin                         1.524209e+10
71          Épices, Autres                         1.770676e+10

[72 rows x 2 columns]

In [ ]:
dispo_veg_mondial= tab_vegetal['Disponibilité alimentaire (Kcal/an)'].sum()
dispo_veg_mondial
print ('la disponibilité mondiale de l\'ensemble des végétaux est de','{:,}'.format(int(dispo_veg_mondial)).replace(',', ' '),
      'Kcal.')

la disponibilité mondiale de l'ensemble des végétaux est de 6 296 671 916 099 Kcal.


In [ ]:
dispo_veg_mondial2=  (dispo_veg_mondial *1000 /912500)
dispo_veg_mondial2

6900462373.807601

In [ ]:
pourcentage_humain_nourris_dispoalim_cal2 = round((dispo_veg_mondial2 / population_mondiale * 100),2)
pourcentage_humain_nourris_dispoalim_cal2

92.12

**Avec la disponibilité des végétaux mondiaux on peut nourrir 92,12% de la population mondiale**

L’utilisation de la disponibilité intérieure, en particulier la part qui est attribuée à l’alimentation animale de celle qui est perdue, de celle qui est concrètement utilisée pour l'alimentation humaine.


In [ ]:
tab_vegetal = tab_vegetal.groupby(by = 'Produit').sum().reset_index()
tab_vegetal

Produit  Aliments pour animaux  Autres Utilisations  \
0          Agrumes, Autres                    0.0                 25.0   
1    Aliments pour enfants                    0.0                  0.0   
2                   Ananas                    0.0                  3.0   
3   Arachides Decortiquees                    1.0               1265.0   
4                   Avoine                15849.0                199.0   
..                     ...                    ...                  ...   
67                  Sésame                    0.0                 19.0   
68                     Thé                    0.0                 26.0   
69                 Tomates                 1870.0                 77.0   
70                     Vin                    0.0               1564.0   
71          Épices, Autres                    0.0                185.0   

    Disponibilité alimentaire (Kcal/personne/jour)  \
0                                            128.0   
1                                            629.0   
2                                            653.0   
3                                           3626.0   
4                                            828.0   
..                                             ...   
67                                           734.0   
68                                           121.0   
69                                          1615.0   
70                                          1970.0   
71                                          1006.0   

    Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                              208.45        
1                                               62.86        
2                                              648.78        
3                                              239.29        
4                                              142.46        
..                                                ...        
67                                              46.68        
68                                             125.89        
69                                            3081.08        
70                                            1031.89        
71                                             106.16        

    Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                                0.78               
1                                                4.88               
2                                                1.88               
3                                              301.37               
4                                               14.34               
..                                                ...               
67                                              64.33               
68                                               0.00               
69                                              18.34               
70                                               0.00               
71                                              37.39               

    Disponibilité de protéines en quantité (g/personne/jour)  \
0                                                2.19          
1                                               26.20          
2                                                4.66          
3                                              158.33          
4                                               33.65          
..                                                ...          
67                                              22.80          
68                                              33.20          
69                                              77.11          
70                                               0.01          
71                                              31.06          

    Disponibilité intérieure  Exportations - Quantité  \
0                    12213.0                    760.0   
1                      6

#**Question  : En ne prenant en compte que les céréales destinées à l'alimentation (humaine et animale), quelle proportion est destinée à l'alimentation animale ?**

In [ ]:
cereales = pd.read_csv ('cereales.csv')
cereales.head()

Code Domaine                      Domaine  Code zone (FAO)         Zone  \
0          FBS  Bilans Alimentaires (2010-)                2  Afghanistan   
1          FBS  Bilans Alimentaires (2010-)                2  Afghanistan   
2          FBS  Bilans Alimentaires (2010-)                2  Afghanistan   
3          FBS  Bilans Alimentaires (2010-)                2  Afghanistan   
4          FBS  Bilans Alimentaires (2010-)                2  Afghanistan   

   Code Élément                Élément  Code Produit           Produit  \
0          5521  Aliments pour animaux          2511               Blé   
1          5142             Nourriture          2511               Blé   
2          5521  Aliments pour animaux          2807  Riz (Eq Blanchi)   
3          5142             Nourriture          2807  Riz (Eq Blanchi)   
4          5521  Aliments pour animaux          2513              Orge   

   Code année  Année               Unité  Valeur Symbole  \
0        2017   2017  Milliers de tonnes     221      Im   
1        2017   2017  Milliers de tonnes    5944      Im   
2        2017   2017  Milliers de tonnes       4      Im   
3        2017   2017  Milliers de tonnes     700      Im   
4        2017   2017  Milliers de tonnes      67      Im   

                              Description du Symbole  
0  Données de la FAO basées sur une méthodologie ...  
1  Données de la FAO basées sur une méthodologie ...  
2  Données de la FAO basées sur une méthodologie ...  
3  Données de la FAO basées sur une méthodologie ...  
4  Données de la FAO basées sur une méthodologie ...

In [ ]:

 liste_cereales = ["Blé et produits", "Riz et produits", "Orge et produits", "Maïs et produits", "Seigle et produits",
                  "Avoine", "Millet et produits", "Sorgho et produits", "Céréales, Autres"]

In [ ]:
filt_bilan_alimentaire['cereal'] = bilan_alimentaire['Produit'].isin(liste_cereales)
filt_bilan_alimentaire.sample(10)


/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


Zone                  Produit   Origine  \
54508              Mexique           Fruits, Autres  vegetale   
4432             Argentine   Huile de Germe de Maïs  vegetale   
33244            Guatemala             Pamplemousse  vegetale   
42478                Japon                     Noix  vegetale   
19828  Chine, continentale  Huil Plantes Oleif Autr  vegetale   
37438                 Inde                     Maïs  vegetale   
87598                Yémen                   Avoine  vegetale   
26860              Estonie                  Bananes  vegetale   
5350             Australie                  Bananes  vegetale   
1366               Albanie                   Pommes  vegetale   

       Aliments pour animaux  Autres Utilisations  \
54508                    NaN                  NaN   
4432                     NaN                  NaN   
33244                    NaN                  NaN   
42478                    NaN                  NaN   
19828                    NaN                915.0   
37438                 6990.0                 45.0   
87598                    0.0                  NaN   
26860                    NaN                  NaN   
5350                     NaN                  NaN   
1366                     NaN                  NaN   

       Disponibilité alimentaire (Kcal/personne/jour)  \
54508                                            32.0   
4432                                             15.0   
33244                                             0.0   
42478                                            12.0   
19828                                             2.0   
37438                                            52.0   
87598                                             0.0   
26860                                            13.0   
5350                                             25.0   
1366                                             33.0   

       Disponibilité alimentaire en quantité (kg/personne/an)  \
54508                                              31.47        
4432                                                0.61        
33244                                               0.04        
42478                                               1.88        
19828                                               0.08        
37438                                               6.29        
87598                                               0.05        
26860                                               7.76        
5350                                               14.85        
1366                                               24.60        

       Disponibilité de matière grasse en quantité (g/personne/jour)  \
54508                                               0.70               
4432                                                1.68               
33244                                                NaN               
42478                                               1.05               
19828                                               0.21               
37438                                               0.64               
87598                                               0.00               
26860                                               0.06               
5350                                                0.12               
1366                                                0.27               

       Disponibilité de protéines en quantité (g/personne/jour)  \
54508                                               0.46          
4432                                                 NaN          
33244                                               0.00          
42478                                               0.34          
19828                                               0.00          
37438                                               1.29          
87598                                               0.01          
26860                                            

##utilisation des céréales

In [ ]:
part_cereales_animales = filt_bilan_alimentaire[filt_bilan_alimentaire['cereal'] == True]['Aliments pour animaux'].sum() 
part_nourriture_humaine = filt_bilan_alimentaire[filt_bilan_alimentaire['cereal'] == True]['Nourriture'].sum()
part_dispo = filt_bilan_alimentaire[filt_bilan_alimentaire['cereal'] == True]['Disponibilité intérieure'].sum()



In [ ]:
part_dispo

49481.0

In [ ]:
part_cereales_animales

34316.0

In [ ]:
part_nourriture_humaine

8767.0

In [ ]:
total_part_cereale = part_cereales_animales + part_nourriture_humaine
total_part_cereale

43083.0

In [ ]:
total_dispo_interieur= filt_bilan_alimentaire['Disponibilité intérieure'].sum()
total_dispo_interieur

8763134.0

In [ ]:
Tx_nourriture1 = (part_nourriture_humaine/part_dispo )*100
Tx_nourriture1

17.717911925789696

**Il est constaté que 17,7% de la disponibilité intérieure est à usage de la nourriture**

In [ ]:
Tx_animal =  (part_cereales_animales / part_dispo)*100 
Tx_animal

69.35187243588447

**il est constaté que 69,35% de la disponibilité intérieure est à usage animale** 


##Production + Importations - Exportations + Variation de stock = Disponibilité intérieure = Semences + Pertes + Nourriture + Aliments pour animaux + Traitement + Autres utilisations


#**Question : En Thaïlande, quelle proportion de manioc est exportée ? Quelle est la proportion de personnes en sous-nutrition ?**

In [ ]:
thai_manioc = filt_bilan_alimentaire[(filt_bilan_alimentaire['Zone'] == 'Thaïlande') & (filt_bilan_alimentaire['Produit'] == 'Manioc')]
thai_manioc = thai_manioc[['Zone', 'Produit', 'Exportations - Quantité','Disponibilité intérieure','Valeur','Production']]
thai_manioc

Zone Produit  Exportations - Quantité  Disponibilité intérieure  \
81724  Thaïlande  Manioc                  25214.0                    6264.0   

         Valeur  Production  
81724  69209.81     30228.0

In [ ]:
thailande_manioc_export = (thai_manioc['Exportations - Quantité'] 
                          / (thai_manioc['Production'] )
                          ) * 100
thailande_manioc_export

81724    83.41273
dtype: float64

**La quantité de manioc exporté est de l'ordre de 83,41%**

In [ ]:
tx_sous_nutris_thailande = (float(sousalimentation_annee[sousalimentation_annee['Zone'] == 'Thaïlande']['2016-2018'])
                            / float(thai_manioc['Valeur']/1000))*100
tx_sous_nutris_thailande

8.958267621309753

**La part de la population considérée comme sous alimenté est de 8,95%**

In [ ]:
filt_bilan_alimentaire=pd.merge(filt_bilan_alimentaire, sousalimentation_annee)
filt_bilan_alimentaire

Zone                Produit   Origine  Aliments pour animaux  \
0      Afghanistan       Abats Comestible   animale                    NaN   
1      Afghanistan        Agrumes, Autres  vegetale                    NaN   
2      Afghanistan  Aliments pour enfants  vegetale                    NaN   
3      Afghanistan                Bananes  vegetale                    NaN   
4      Afghanistan           Beurre, Ghee   animale                    NaN   
...            ...                    ...       ...                    ...   
10239     Éthiopie       Viande de Suides   animale                    NaN   
10240     Éthiopie    Viande de Volailles   animale                    NaN   
10241     Éthiopie          Viande, Autre   animale                    NaN   
10242     Éthiopie                    Vin  vegetale                    NaN   
10243     Éthiopie         Épices, Autres  vegetale                    NaN   

       Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                      NaN                                             5.0   
1                      NaN                                             1.0   
2                      NaN                                             1.0   
3                      NaN                                             4.0   
4                      NaN                                            23.0   
...                    ...                                             ...   
10239                  NaN                                             0.0   
10240                  NaN                                             2.0   
10241                  NaN                                             4.0   
10242                  NaN                                             0.0   
10243                  NaN                                             3.0   

       Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                                   1.72        
1                                                   1.29        
2                                                   0.06        
3                                                   2.70        
4                                                   1.17        
...                                                  ...        
10239                                               0.02        
10240                                               0.66        
10241                                               1.20        
10242                                               0.02        
10243                                               0.29        

       Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                                   0.20               
1                                                   0.01               
2                                                   0.01               
3                                                   0.02               
4                                                   2.61               
...                                                  ...               
10239                                               0.02               
10240                                               0.14               
10241                                               0.18               
10242                                                NaN               
10243                                               0.12               

       Disponibilité de protéines en quantité (g/personne/jour)  \
0                                                   0.77          
1                                                   0.02          
2                                                   0.03          
3                                                   0.05          
4                                                   0.03          
...                                                  ...          
10239                    

In [ ]:
filt_bilan_alimentaire['ratio']= (filt_bilan_alimentaire['2016-2018']/(filt_bilan_alimentaire['Valeur']/1000))*100
filt_bilan_alimentaire

Zone                Produit   Origine  Aliments pour animaux  \
0      Afghanistan       Abats Comestible   animale                    NaN   
1      Afghanistan        Agrumes, Autres  vegetale                    NaN   
2      Afghanistan  Aliments pour enfants  vegetale                    NaN   
3      Afghanistan                Bananes  vegetale                    NaN   
4      Afghanistan           Beurre, Ghee   animale                    NaN   
...            ...                    ...       ...                    ...   
10239     Éthiopie       Viande de Suides   animale                    NaN   
10240     Éthiopie    Viande de Volailles   animale                    NaN   
10241     Éthiopie          Viande, Autre   animale                    NaN   
10242     Éthiopie                    Vin  vegetale                    NaN   
10243     Éthiopie         Épices, Autres  vegetale                    NaN   

       Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                      NaN                                             5.0   
1                      NaN                                             1.0   
2                      NaN                                             1.0   
3                      NaN                                             4.0   
4                      NaN                                            23.0   
...                    ...                                             ...   
10239                  NaN                                             0.0   
10240                  NaN                                             2.0   
10241                  NaN                                             4.0   
10242                  NaN                                             0.0   
10243                  NaN                                             3.0   

       Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                                   1.72        
1                                                   1.29        
2                                                   0.06        
3                                                   2.70        
4                                                   1.17        
...                                                  ...        
10239                                               0.02        
10240                                               0.66        
10241                                               1.20        
10242                                               0.02        
10243                                               0.29        

       Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                                   0.20               
1                                                   0.01               
2                                                   0.01               
3                                                   0.02               
4                                                   2.61               
...                                                  ...               
10239                                               0.02               
10240                                               0.14               
10241                                               0.18               
10242                                                NaN               
10243                                               0.12               

       Disponibilité de protéines en quantité (g/personne/jour)  \
0                                                   0.77          
1                                                   0.02          
2                                                   0.03          
3                                                   0.05          
4                                                   0.03          
...                                                  ...          
10239                    

In [ ]:
filt_bilan_alimentaire1=filt_bilan_alimentaire.groupby('Zone').sum()
filt_bilan_alimentaire1

Aliments pour animaux  Autres Utilisations  \
Zone                                                                
Afghanistan                            687.0                415.0   
Afrique du Sud                        5308.0                726.0   
Albanie                                659.0                168.0   
Algérie                               4350.0               1701.0   
Allemagne                            29931.0               6314.0   
...                                      ...                  ...   
Égypte                               14880.0               1552.0   
Émirats arabes unis                    626.0               2215.0   
Équateur                              1057.0               1126.0   
États-Unis d'Amérique               146238.0             152853.0   
Éthiopie                               685.0               3641.0   

                       Disponibilité alimentaire (Kcal/personne/jour)  \
Zone                                                                    
Afghanistan                                                    2086.0   
Afrique du Sud                                                 3020.0   
Albanie                                                        3183.0   
Algérie                                                        3291.0   
Allemagne                                                      3499.0   
...                                                               ...   
Égypte                                                         3513.0   
Émirats arabes unis                                            3267.0   
Équateur                                                       2343.0   
États-Unis d'Amérique                                          3679.0   
Éthiopie                                                       2129.0   

                       Disponibilité alimentaire en quantité (kg/personne/an)  \
Zone                                                                            
Afghanistan                                                       351.41        
Afrique du Sud                                                    564.80        
Albanie                                                          1095.54        
Algérie                                                           809.13        
Allemagne                                                         957.52        
...                                                                  ...        
Égypte                                                            803.79        
Émirats arabes unis                                               700.24        
Équateur                                                          544.91        
États-Unis d'Amérique                                            1002.39        
Éthiopie                                                          354.55        

                       Disponibilité de matière grasse en quantité (g/personne/jour)  \
Zone                                                                                   
Afghanistan                                                        33.48               
Afrique du Sud                                                     83.58               
Albanie                                                           106.42               
Algérie                                                            76.45               
Allemagne                                                         141.62               
...                                                                  ...               
Égypte                                                             63.06               
Émirats arabes unis                                                87.06               
Équateur                                                           94.62               
États-Unis d'Amérique                                             161.37               
Éthiopie                                                         

In [ ]:
filt_bilan_alimentaire1.sort_values("ratio",axis=0,ascending =True, inplace = True)
filt_bilan_alimentaire1.head(10)


Aliments pour animaux  Autres Utilisations  \
Zone                                                      
Tadjikistan                 2588.0                121.0   
Kazakhstan                  6938.0               1556.0   
Japon                      15591.0               2192.0   
Italie                     15624.0               2874.0   
Israël                      2434.0                772.0   
Islande                       65.0                 20.0   
Irlande                     3750.0                963.0   
Zambie                       408.0                 63.0   
Tunisie                     1303.0                157.0   
Hongrie                     4585.0                144.0   

             Disponibilité alimentaire (Kcal/personne/jour)  \
Zone                                                          
Tadjikistan                                          2199.0   
Kazakhstan                                           3254.0   
Japon                                                2721.0   
Italie                                               3577.0   
Israël                                               3608.0   
Islande                                              3356.0   
Irlande                                              3600.0   
Zambie                                               1923.0   
Tunisie                                              3347.0   
Hongrie                                              3032.0   

             Disponibilité alimentaire en quantité (kg/personne/an)  \
Zone                                                                  
Tadjikistan                                             520.41        
Kazakhstan                                              994.14        
Japon                                                   602.30        
Italie                                                  993.20        
Israël                                                  922.14        
Islande                                                 923.29        
Irlande                                                1150.15        
Zambie                                                  350.12        
Tunisie                                                 840.13        
Hongrie                                                 720.36        

             Disponibilité de matière grasse en quantité (g/personne/jour)  \
Zone                                                                         
Tadjikistan                                              60.37               
Kazakhstan                                              134.37               
Japon                                                    86.43               
Italie                                                  154.69               
Israël                                                  151.39               
Islande                                                 151.04               
Irlande                                                 125.69               
Zambie                                                   42.04               
Tunisie                                                  91.44               
Hongrie                                                 137.46               

             Disponibilité de protéines en quantité (g/personne/jour)  \
Zone                                                                    
Tadjikistan                                              63.38          
Kazakhstan                                               96.16          
Japon                                                    87.53          
Italie                                                  108.48          
Israël                                                  128.00          
Islande                                                 132.33          
Irlande                                                 109.91          
Zambie                                                   55.20          
Tunisie                    

**L'ordre Décroissant des pays ayant le plus fort ratio quant à la sous nutrition et la population**

In [ ]:
aide_alimentaire.head()

Pays bénéficiaire  Année              Produit  Valeur
0       Afghanistan   2013  Autres non-céréales     682
1       Afghanistan   2014  Autres non-céréales     335
2       Afghanistan   2013         Blé et Farin   39224
3       Afghanistan   2014         Blé et Farin   15160
4       Afghanistan   2013             Céréales   40504

In [ ]:
aide_alimentaire1=aide_alimentaire
aide_alimentaire1=aide_alimentaire1.groupby( 'Pays bénéficiaire').sum().reset_index()[['Pays bénéficiaire','Valeur']]
aide_alimentaire1

Pays bénéficiaire   Valeur
0        Afghanistan   185452
1            Algérie    81114
2             Angola     5014
3         Bangladesh   348188
4            Bhoutan     2666
..               ...      ...
71            Zambie     3026
72          Zimbabwe    62570
73            Égypte     1122
74          Équateur     1362
75          Éthiopie  1381294

[76 rows x 2 columns]

In [ ]:
aide_alimentaire1.sort_values(by="Valeur",ascending=False,inplace=True)

In [ ]:
aide_alimentaire1.head(10)

Pays bénéficiaire   Valeur
50         République arabe syrienne  1858943
75                          Éthiopie  1381294
70                             Yémen  1206484
61                     Soudan du Sud   695248
60                            Soudan   669784
30                             Kenya   552836
3                         Bangladesh   348188
59                           Somalie   292678
53  République démocratique du Congo   288502
43                             Niger   276344

In [ ]:
aide_alimentaire1.sort_values("Valeur",axis=0,ascending =False, inplace = True)

In [ ]:
aide_alimentaire1.head(10)

Pays bénéficiaire   Valeur
50         République arabe syrienne  1858943
75                          Éthiopie  1381294
70                             Yémen  1206484
61                     Soudan du Sud   695248
60                            Soudan   669784
30                             Kenya   552836
3                         Bangladesh   348188
59                           Somalie   292678
53  République démocratique du Congo   288502
43                             Niger   276344

**Liste des pays ayant eu le plus d'aide alimentaire depuis 2013**

In [ ]:
filt_bilan_alimentaire['ratio disponibilité']= (filt_bilan_alimentaire['Disponibilité alimentaire (Kcal/an)']
                                                /(filt_bilan_alimentaire['Valeur']/1000)*100)
filt_bilan_alimentaire

Zone                Produit   Origine  Aliments pour animaux  \
0      Afghanistan       Abats Comestible   animale                    NaN   
1      Afghanistan        Agrumes, Autres  vegetale                    NaN   
2      Afghanistan  Aliments pour enfants  vegetale                    NaN   
3      Afghanistan                Bananes  vegetale                    NaN   
4      Afghanistan           Beurre, Ghee   animale                    NaN   
...            ...                    ...       ...                    ...   
10239     Éthiopie       Viande de Suides   animale                    NaN   
10240     Éthiopie    Viande de Volailles   animale                    NaN   
10241     Éthiopie          Viande, Autre   animale                    NaN   
10242     Éthiopie                    Vin  vegetale                    NaN   
10243     Éthiopie         Épices, Autres  vegetale                    NaN   

       Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                      NaN                                             5.0   
1                      NaN                                             1.0   
2                      NaN                                             1.0   
3                      NaN                                             4.0   
4                      NaN                                            23.0   
...                    ...                                             ...   
10239                  NaN                                             0.0   
10240                  NaN                                             2.0   
10241                  NaN                                             4.0   
10242                  NaN                                             0.0   
10243                  NaN                                             3.0   

       Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                                   1.72        
1                                                   1.29        
2                                                   0.06        
3                                                   2.70        
4                                                   1.17        
...                                                  ...        
10239                                               0.02        
10240                                               0.66        
10241                                               1.20        
10242                                               0.02        
10243                                               0.29        

       Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                                   0.20               
1                                                   0.01               
2                                                   0.01               
3                                                   0.02               
4                                                   2.61               
...                                                  ...               
10239                                               0.02               
10240                                               0.14               
10241                                               0.18               
10242                                                NaN               
10243                                               0.12               

       Disponibilité de protéines en quantité (g/personne/jour)  \
0                                                   0.77          
1                                                   0.02          
2                                                   0.03          
3                                                   0.05          
4                                                   0.03          
...                                                  ...          
10239                    

In [ ]:
filt_bilan_alimentaire2=filt_bilan_alimentaire.groupby('Zone').sum()
filt_bilan_alimentaire2

Aliments pour animaux  Autres Utilisations  \
Zone                                                                
Afghanistan                            687.0                415.0   
Afrique du Sud                        5308.0                726.0   
Albanie                                659.0                168.0   
Algérie                               4350.0               1701.0   
Allemagne                            29931.0               6314.0   
...                                      ...                  ...   
Égypte                               14880.0               1552.0   
Émirats arabes unis                    626.0               2215.0   
Équateur                              1057.0               1126.0   
États-Unis d'Amérique               146238.0             152853.0   
Éthiopie                               685.0               3641.0   

                       Disponibilité alimentaire (Kcal/personne/jour)  \
Zone                                                                    
Afghanistan                                                    2086.0   
Afrique du Sud                                                 3020.0   
Albanie                                                        3183.0   
Algérie                                                        3291.0   
Allemagne                                                      3499.0   
...                                                               ...   
Égypte                                                         3513.0   
Émirats arabes unis                                            3267.0   
Équateur                                                       2343.0   
États-Unis d'Amérique                                          3679.0   
Éthiopie                                                       2129.0   

                       Disponibilité alimentaire en quantité (kg/personne/an)  \
Zone                                                                            
Afghanistan                                                       351.41        
Afrique du Sud                                                    564.80        
Albanie                                                          1095.54        
Algérie                                                           809.13        
Allemagne                                                         957.52        
...                                                                  ...        
Égypte                                                            803.79        
Émirats arabes unis                                               700.24        
Équateur                                                          544.91        
États-Unis d'Amérique                                            1002.39        
Éthiopie                                                          354.55        

                       Disponibilité de matière grasse en quantité (g/personne/jour)  \
Zone                                                                                   
Afghanistan                                                        33.48               
Afrique du Sud                                                     83.58               
Albanie                                                           106.42               
Algérie                                                            76.45               
Allemagne                                                         141.62               
...                                                                  ...               
Égypte                                                             63.06               
Émirats arabes unis                                                87.06               
Équateur                                                           94.62               
États-Unis d'Amérique                                             161.37               
Éthiopie                                                         

In [ ]:
filt_bilan_alimentaire2.sort_values("ratio disponibilité",axis=0, ascending=True, inplace=True)

#Liste des pays ayant la plus faible disponibilité par habitants

In [ ]:
filt_bilan_alimentaire2

Aliments pour animaux  Autres Utilisations  \
Zone                                                                    
République centrafricaine                    1.0                 19.0   
Zambie                                     408.0                 63.0   
Madagascar                                 842.0               2509.0   
Afghanistan                                687.0                415.0   
Haïti                                      481.0                110.0   
...                                          ...                  ...   
Israël                                    2434.0                772.0   
États-Unis d'Amérique                   146238.0             152853.0   
Turquie                                  11467.0               2775.0   
Belgique                                  5117.0               2979.0   
Autriche                                  3400.0                992.0   

                           Disponibilité alimentaire (Kcal/personne/jour)  \
Zone                                                                        
République centrafricaine                                          1878.0   
Zambie                                                             1923.0   
Madagascar                                                         2056.0   
Afghanistan                                                        2086.0   
Haïti                                                              2089.0   
...                                                                   ...   
Israël                                                             3608.0   
États-Unis d'Amérique                                              3679.0   
Turquie                                                            3708.0   
Belgique                                                           3732.0   
Autriche                                                           3767.0   

                           Disponibilité alimentaire en quantité (kg/personne/an)  \
Zone                                                                                
République centrafricaine                                             453.73        
Zambie                                                                350.12        
Madagascar                                                            425.07        
Afghanistan                                                           351.41        
Haïti                                                                 397.78        
...                                                                      ...        
Israël                                                                922.14        
États-Unis d'Amérique                                                1002.39        
Turquie                                                               968.15        
Belgique                                                             1002.09        
Autriche                                                             1069.20        

                           Disponibilité de matière grasse en quantité (g/personne/jour)  \
Zone                                                                                       
République centrafricaine                                              59.40               
Zambie                                                                 42.04               
Madagascar                                                             24.48               
Afghanistan                                                            33.48               
Haïti                                                                  48.91               
...                                                                      ...               
Israël                                                                151.39               
États-Unis d'Amérique                                                 161.37               
Turquie                                          

In [ ]:
filt_bilan_alimentaire2.sort_values("ratio disponibilité",axis=0, ascending=False, inplace=True)

#Liste des pays ayant la plus grande disponibilté par habitants 

In [ ]:
filt_bilan_alimentaire2

Aliments pour animaux  Autres Utilisations  \
Zone                                                                    
Autriche                                  3400.0                992.0   
Belgique                                  5117.0               2979.0   
Turquie                                  11467.0               2775.0   
États-Unis d'Amérique                   146238.0             152853.0   
Israël                                    2434.0                772.0   
...                                          ...                  ...   
Haïti                                      481.0                110.0   
Afghanistan                                687.0                415.0   
Madagascar                                 842.0               2509.0   
Zambie                                     408.0                 63.0   
République centrafricaine                    1.0                 19.0   

                           Disponibilité alimentaire (Kcal/personne/jour)  \
Zone                                                                        
Autriche                                                           3767.0   
Belgique                                                           3732.0   
Turquie                                                            3708.0   
États-Unis d'Amérique                                              3679.0   
Israël                                                             3608.0   
...                                                                   ...   
Haïti                                                              2089.0   
Afghanistan                                                        2086.0   
Madagascar                                                         2056.0   
Zambie                                                             1923.0   
République centrafricaine                                          1878.0   

                           Disponibilité alimentaire en quantité (kg/personne/an)  \
Zone                                                                                
Autriche                                                             1069.20        
Belgique                                                             1002.09        
Turquie                                                               968.15        
États-Unis d'Amérique                                                1002.39        
Israël                                                                922.14        
...                                                                      ...        
Haïti                                                                 397.78        
Afghanistan                                                           351.41        
Madagascar                                                            425.07        
Zambie                                                                350.12        
République centrafricaine                                             453.73        

                           Disponibilité de matière grasse en quantité (g/personne/jour)  \
Zone                                                                                       
Autriche                                                              170.40               
Belgique                                                              161.95               
Turquie                                                               119.64               
États-Unis d'Amérique                                                 161.37               
Israël                                                                151.39               
...                                                                      ...               
Haïti                                                                  48.91               
Afghanistan                                                            33.48               
Madagascar                                       